isthmus/src/test/java/io/substrait/isthmus/ApplyJoinPlanTest.java

```diff
@@ -0,0 +1,125 @@
+package io.substrait.isthmus;
+
+import java.util.Map;
+import org.apache.calcite.adapter.tpcds.TpcdsSchema;
+import org.apache.calcite.rel.RelRoot;
+import org.apache.calcite.rex.RexFieldAccess;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.sql.validate.SqlConformanceEnum;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class ApplyJoinPlanTest {
+
+  private static RelRoot getCalcitePlan(SqlToSubstrait s, TpcdsSchema schema, String sql)
+      throws SqlParseException {
+    var pair = s.registerSchema("tpcds", schema);
+    var converter = s.createSqlToRelConverter(pair.left, pair.right);
+    SqlParser parser = SqlParser.create(sql, s.parserConfig);
+    var root = s.getBestExpRelRoot(converter, parser.parseQuery());
+    return root;
+  }
+
+  private static void validateOuterRef(
```

Comment:
In addition to validate `$cor0.SS_ITEM_SK` is in the map, is it possible to verify the depth of the correlated variable?  

Also, the test seems to have a one level nested correlated variable (depth = 1).  Does it make sense to have multi-level correlated variable (i.e. depth = 2) and verify it works as expected?

Reply:
Thanks again for the suggestion. In the latest update I have added test for nested-apply-join and added depth verification too.